pkg/cloudprovider/cloudprovider.go

```diff
@@ -178,18 +178,19 @@ func (c *Client) UpdateUserMSI(addUserAssignedMSIIDs, removeUserAssignedMSIIDs [
 	}
 
 	ids := make(map[string]bool)
-	// add new ids to the list
-	for _, userAssignedMSIID := range addUserAssignedMSIIDs {
-		ids[userAssignedMSIID] = true
-	}
 	// remove msi ids from the list
 	for _, userAssignedMSIID := range removeUserAssignedMSIIDs {
 		ids[userAssignedMSIID] = false
 	}
+	// add new ids to the list
+	for _, userAssignedMSIID := range addUserAssignedMSIIDs {
```

Comment:
Is there. change here ?

Reply:
Yeah, the remove needs to be done before assign. The order is required to ensure we don't delete an identity that needs to be assigned.

Reply:
This is part of #564 now - right ?

Reply:
Yeah, rebased the PR now on the master.